File: VuongVL/app/src/main/java/com/example/vuongvl/MainActivity.java
package com.example.vuongvl;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements ISpam {
    ServiceMail serviceMail;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        serviceMail = new ServiceMail(this);
        ArrayList<Client> mClient = new ArrayList<>();
        Data(mClient);
        serviceMail.registerSpam(mClient);
    }

    private void Data(ArrayList<Client> mClient) {
        mClient.add(new Client("Vo", true));
        mClient.add(new Client("Long", false));
        mClient.add(new Client("Vuong", false));
        mClient.add(new Client("Hung", true));
    }

    @Override
    public void spamMail(String name) {
        Log.d("thongtin", "mail Spam duoc goi toi cho khach hang: " + name);
    }
}

File: VuongVLKotlin/app/src/main/java/com/example/vuongvlkotlin/MainActivity.kt
package com.example.vuongvlkotlin

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import java.util.*

class MainActivity : AppCompatActivity() {
    var serviceMail: ServiceMail? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        serviceMail = ServiceMail()
        val mClient = ArrayList<Client>()
        Data(mClient)
        serviceMail!!.registerSpam(mClient)
    }

    private fun Data(mClient: ArrayList<Client>) {
        mClient.add(Client("Vo", true))
        mClient.add(Client("Long", false))
        mClient.add(Client("Vuong", false))
        mClient.add(Client("Hung", true))
    }

    fun spamMail(name: String) {
        Log.d("thongtin", "mail Spam duoc goi toi cho khach hang: $name")
    }
}